Lib/pdb.py

```diff
@@ -436,6 +438,20 @@ def _cmdloop(self):
             except KeyboardInterrupt:
                 self.message('--KeyboardInterrupt--')
 
+    def _validate_file_mtime(self):
+        """Check if the source file of the current frame has been modified since
+        the last time we saw it. If so, give a warning."""
+        try:
+            filename = self.curframe.f_code.co_filename
+            mtime = os.path.getmtime(filename)
+        except Exception:
+            return
+        if (filename in self._file_mtime_table and
+            mtime != self._file_mtime_table[filename]):
+            self.message(f"*** WARNING: file '{filename}' was edited after pdb started, "
+                         "running stale code until restart")
```

Comment:
Maybe "until program is rerun" actually. I think this could look like "until restart of pdb".

Reply:
And also, "after pdb restarted" is not quite accurate, right?

Reply:
It's not the full truth but it's accurate. If the user sees this warning, then the file must be edited after pdb started. However, it's possible that the file was edited after the pdb started, but the warning was not generated - because of how we detect this kind of issue.